plugins/Diagnostics/DiagnosticReport.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Piwik\Plugins\Diagnostics;
+
+use Piwik\Plugins\Diagnostics\Diagnostic\DiagnosticResult;
+
+/**
+ * A diagnostic report contains all the results of all the diagnostics.
+ */
+class DiagnosticReport
+{
+    /**
+     * @var DiagnosticResult[]
+     */
+    private $mandatoryDiagnosticResults;
+
+    /**
+     * @var DiagnosticResult[]
+     */
+    private $optionalDiagnosticResults;
+
+    /**
+     * @var int|null
+     */
+    private $errorCount;
+
+    /**
+     * @var int|null
+     */
+    private $warningCount;
+
+    /**
+     * @param DiagnosticResult[] $mandatoryDiagnosticResults
+     * @param DiagnosticResult[] $optionalDiagnosticResults
+     */
+    public function __construct(array $mandatoryDiagnosticResults, array $optionalDiagnosticResults)
+    {
+        $this->mandatoryDiagnosticResults = $mandatoryDiagnosticResults;
+        $this->optionalDiagnosticResults = $optionalDiagnosticResults;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasErrors()
+    {
+        return $this->getErrorCount() > 0;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasWarnings()
+    {
+        return $this->getWarningCount() > 0;
+    }
+
+    /**
+     * @return int
+     */
+    public function getErrorCount()
+    {
+        if ($this->errorCount === null) {
+            $this->errorCount = 0;
+            foreach ($this->getAllResults() as $result) {
+                if ($result->getStatus() === DiagnosticResult::STATUS_ERROR) {
+                    $this->errorCount++;
+                }
+            }
+        }
```

Comment:
Instead of caching on demand, do you think it might be better to compute this upon construction? Not an important thing, also not sure whether it would really be better.

Same for getWarningCount().